tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -474,14 +493,15 @@ class HWCommandQueue {
     void enqueue_record_event(std::reference_wrapper<Event> event);
     void populate_record_event(std::reference_wrapper<Event> event);
     void enqueue_wait_for_event(std::reference_wrapper<Event> event);
+    void enqueue_trace();
     void finish();
     void issue_wrap();
     void completion_wrap(uint32_t event);
     void launch(launch_msg_t& msg);
-    friend void EnqueueTrace(Trace& trace, bool blocking);
-    friend void EndTrace(Trace& trace);
-    friend Trace BeginTrace(CommandQueue& cq);
-    friend void EnqueueProgramImpl(CommandQueue& cq, std::variant < std::reference_wrapper<Program>, std::shared_ptr<Program> > program, bool blocking, std::optional<std::reference_wrapper<Trace>> trace);
+    friend void EnqueueTraceImpl(CommandQueue& cq);
+    // friend void EndTrace(Trace& trace);
+    // friend Trace BeginTrace(CommandQueue& cq);
```

Comment:
We can just get rid of these, unless we expect that theses needs to access private CQ members later on

Reply:
Thanks, got rid of them in the latest commit